options.c

```diff
@@ -441,37 +443,81 @@ GetConnByName(const WCHAR *name)
     return NULL;
 }
 
-/* callback to set the initial value of folder browse selection */
-static int CALLBACK
-BrowseCallback (HWND h, UINT msg, UNUSED LPARAM l, LPARAM data)
+static BOOL
+BrowseFolder (const WCHAR* initial_path, WCHAR* selected_path)
 {
-    if (msg == BFFM_INITIALIZED)
-        SendMessage (h, BFFM_SETSELECTION, TRUE, data);
+    IFileOpenDialog* pfd;
+    HRESULT initResult, result, dialogResult;
 
-    return 0;
-}
+    // Create dialog
+    initResult = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED);
+    if (FAILED(initResult))
+    {
+        return false;
+    }
 
-static BOOL
-BrowseFolder (const WCHAR *initial_path, WCHAR *selected_path)
-{
-    BOOL ret = false;
-    BROWSEINFO bi;
+    result = CoCreateInstance(&CLSID_FileOpenDialog, NULL, CLSCTX_ALL, &IID_IFileOpenDialog, &pfd);
+    if (SUCCEEDED(result))
+    {
+        // Select folders, not files
+        {
+            DWORD dwOptions;
+            result = pfd->lpVtbl->GetOptions(pfd, &dwOptions);
+            if(SUCCEEDED(result))
+            {
+                dwOptions |= FOS_PICKFOLDERS;
+                result = pfd->lpVtbl->SetOptions(pfd, dwOptions);
+            }
+        }
 
-    CLEAR(bi);
-    bi.lpszTitle = L"Select folder...";
-    bi.ulFlags = BIF_RETURNONLYFSDIRS | BIF_NEWDIALOGSTYLE;
-    bi.lpfn    = BrowseCallback;
-    bi.lParam  = (LPARAM) initial_path;
+        // Set initial path
+        {
+            IShellItem* psi;
+            result = SHCreateItemFromParsingName(initial_path, NULL, &IID_IShellItem, &psi);
+            if (SUCCEEDED(result))
+            {
+                pfd->lpVtbl->SetDefaultFolder(pfd, psi);
```

Comment:
SetDefaultFolder is not the right one to use here as it does not force the initial path. Use SetFolder. Otherwise, the dialog will always start with the previous folder which is not what we want as we use this for both config and log locations.